File: module/service-grpc/service_lease.go
package main

func leaseGrant(in []byte) (err error) {
	kvShard().AsyncApply(append(in, CMD_LEASE_GRANT), []byte("leaseGrant"))
	return
}

func leaseRevoke(in []byte) (err error) {
	kvShard().AsyncApply(append(in, CMD_LEASE_REVOKE), []byte("leaseRevoke"))
	return
}

func leaseKeepaliveOpen() (err error) {
	return
}

func leaseKeepaliveRecv(item []byte) (err error) {
	kvShard().AsyncApply(append(item, CMD_LEASE_KEEP_ALIVE), []byte(`leaseKeepaliveRecv`))
	return
}

func leaseKeepaliveClose() (err error) {
	return
}

func leaseLeases(in []byte) (err error) {
	kvShard().AsyncRead(append(in, QUERY_LEASE_LEASES), []byte(`leaseLeases`), false)
	return
}

func leaseTimeToLive(in []byte) (err error) {
	kvShard().AsyncRead(append(in, QUERY_LEASE_TIME_TO_LIVE), []byte(`leaseTimeToLive`), false)
	return
}

File: module-zig/service-grpc/service_lease.zig
//! Mirrors module/service-grpc/service_lease.go

const types = @import("types.zig");
const util = @import("util.zig");

pub fn grant(in: []const u8) anyerror!void {
    return util.kvShard().AsyncApply(util.withSuffix(in, types.CMD_LEASE_GRANT), "leaseGrant");
}

pub fn revoke(in: []const u8) anyerror!void {
    return util.kvShard().AsyncApply(util.withSuffix(in, types.CMD_LEASE_REVOKE), "leaseRevoke");
}

pub fn keepaliveOpen() anyerror!void {}

pub fn keepaliveRecv(in: []const u8) anyerror!void {
    return util.kvShard().AsyncApply(util.withSuffix(in, types.CMD_LEASE_KEEP_ALIVE), "leaseKeepAlive");
}

pub fn keepaliveClose() anyerror!void {}

pub fn leases(in: []const u8) anyerror!void {
    return util.kvShard().AsyncRead(util.withSuffix(in, types.QUERY_LEASE_LEASES), "leaseLeases", true);
}

pub fn timeToLive(in: []const u8) anyerror!void {
    return util.kvShard().AsyncRead(util.withSuffix(in, types.QUERY_LEASE_TIME_TO_LIVE), "leaseTimeToLive", true);
}
